app/models/issue_assigner.rb

```diff
@@ -19,28 +19,10 @@ def assign!
   private
 
   def assign_issue_for_sub(sub)
-    issue = Issue.find_by_sql(<<-SQL
-                SELECT
-                  *
-                FROM
-                  issues
-                WHERE
-                  repo_id = '#{sub.repo_id}' and
-                  state   = '#{Issue::OPEN}'
-                  AND id NOT IN (
-                    SELECT
-                      issue_id
-                    FROM
-                      issue_assignments
-                    WHERE
-                      repo_subscription_id = '#{sub.id}'
-                  )
-                ORDER BY
-                  random()
-                LIMIT
-                  1
-                SQL
-                             ).first
+    issue = Issue.where(repo_id: sub.repo_id, state: Issue::OPEN)
```

Comment:
Since you're refactoring this query, the `Issue` model has a "scope" `.open_issues` to use here.

Reply:
That'll be better.